src/psyclone/dynamo0p3.py

```diff
@@ -3240,25 +3240,27 @@ class DynScalarArgs(DynCollection):
     def __init__(self, node):
         super(DynScalarArgs, self).__init__(node)
 
+        # Create lists of real and integer scalar arguments
+        self._real_scalar_names = {}
+        self._int_scalar_names = {}
         if self._invoke:
-            # Get a list of Invoke scalar arguments
-            self._real_scalars = self._invoke.unique_declns_by_intent(
-                "gh_real")
-            self._int_scalars = self._invoke.unique_declns_by_intent(
-                "gh_integer")
+            real_scalars = self._invoke.unique_declns_by_intent("gh_real")
+            int_scalars = self._invoke.unique_declns_by_intent("gh_integer")
+            for intent in FORTRAN_INTENT_NAMES:
+                self._real_scalar_names[intent] = [arg.declaration_name for arg
+                                                   in real_scalars[intent]]
+                self._int_scalar_names[intent] = [arg.declaration_name for arg
+                                                  in int_scalars[intent]]
         else:
-            # We have a Kernel stub, so create a list of scalar arguments.
-            self._real_scalars = {}
-            self._int_scalars = {}
             for intent in FORTRAN_INTENT_NAMES:
```

Comment:
It was moved there. Are you talking about the `for` loop by Fortran intents where the values are initialised to empty lists? If so, I moved that as well, just below the dict initialisation. It seems to be needing both for the kernel stub.

Reply:
Oh yes. I was thinking that we could get rid of the `self._real/int_scalars = {}` but of course we can't. Apologies but would you therefore mind undoing this last change and putting the loop that initialises the various dict keys back so that it's only done for the kernel stub.

Reply:
Nope, we can't get rid of that, I tried :) Undone the last change and put the key initialisation back.